app/executor.php

```diff
@@ -0,0 +1,722 @@
+    ->param('commands', [], new ArrayList(new Text(0)), 'Commands required to build the container')
+    ->param('runtime', '', new Text(128), 'Runtime for the cloud function')
+    ->param('network', '', new Text(128), 'Network to attach the container to')
+    ->param('baseImage', '', new Text(128), 'Base image name of the runtime')
+    ->param('entrypoint', '', new Text(256), 'Entrypoint of the code file', true)
+    ->param('remove', false, new Boolean(), 'Remove a runtime after execution')
+    ->param('workdir', '', new Text(256), 'Working directory', true)
+    ->inject('orchestrationPool')
+    ->inject('activeRuntimes')
+    ->inject('response')
+    ->action(function (string $runtimeId, string $source, string $destination, array $vars, array $commands, string $runtime, string $network, string $baseImage, string $entrypoint, bool $remove, string $workdir, $orchestrationPool, $activeRuntimes, Response $response) {
+
+        if ($activeRuntimes->exists($runtimeId)) {
+            throw new Exception('Runtime already exists.', 409);
+        }
+
+        $container = [];
+        $containerId = '';
+        $stdout = '';
+        $stderr = '';
+        $startTime = \time();
+        $endTime = 0;
+
+        try {
+            Console::info('Building container : ' . $runtimeId);
+            /** 
+             * Temporary file paths in the executor 
+             */
+            
+            $tmpSource = "/tmp/$runtimeId/src/code.tar.gz";
+            $tmpBuild = "/tmp/$runtimeId/builds/code.tar.gz";
+
+            /**
+             * Copy code files from source to a temporary location on the executor
+             */
+            $sourceDevice = getStorageDevice("/");
+            $localDevice = new Local();
+            $buffer = $sourceDevice->read($source);
+            if(!$localDevice->write($tmpSource, $buffer)) {
+                throw new Exception('Failed to copy source code to temporary directory', 500);
+            };
+
+            /**
+             * Create the mount folder
+             */
+            if (!\file_exists(\dirname($tmpBuild))) {
+                if (!@\mkdir(\dirname($tmpBuild), 0755, true)) {
+                    throw new Exception("Failed to create temporary directory", 500);
+                }
+            }
+
+            /**
+             * Create container
+             */
+            $orchestration = $orchestrationPool->get();
+            $secret = \bin2hex(\random_bytes(16));
+            $vars = \array_merge($vars, [
+                'INTERNAL_RUNTIME_KEY' => $secret,
+                'INTERNAL_RUNTIME_ENTRYPOINT' => $entrypoint,
+            ]);
+            $vars = array_map(fn ($v) => strval($v), $vars);
+            $orchestration
+                ->setCpus(App::getEnv('_APP_FUNCTIONS_CPUS', 1))
+                ->setMemory(App::getEnv('_APP_FUNCTIONS_MEMORY', 256))
+                ->setSwap(App::getEnv('_APP_FUNCTIONS_MEMORY_SWAP', 256));
```

Comment:
Some linux setups have issues while providing `CPU` and `SWAP` option, haven't encountered similar thing for Memory but it might as well cause issue, so there must be a way to disable this options - easiest was , if the environment variable is empty the we don't provide default limit. Recently got this on a digitalocean droplet

When a swap limit is set to something like 2000
```
Docker Error: WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
```
(this seems to be only warning but seems to cause build to fail)

When I set env for swap empty
```
Utopia\Orchestration\Orchestration::setSwap(): Argument #1 ($mb) must be of type int, string given, called in /usr/src/code/app/executor.php on line 205
```